openquake/smt/comparison/utils_gmpes.py

```diff
@@ -72,34 +42,32 @@ def get_sites_from_rupture(rup, from_point='TC', toward_azimuth=90,
     :return:
         A :class:`openquake.hazardlib.site.SiteCollection` instance
     """
-    from_pnt = _get_first_point(rup, from_point)
-    lon = from_pnt.longitude
-    lat = from_pnt.latitude
-    depth = 0
+    from_pnt = rup.surface._get_top_edge_centroid() # Get up-dip edge centroid
```

Comment:
This could have been done with _get_first_point(rup, 'TC') while keeping the flexibility offered by the above function.